api/v1/verticadb_webhook.go

```diff
@@ -1286,6 +1286,22 @@ func (v *VerticaDB) checkImmutableBasic(oldObj *VerticaDB, allErrs field.ErrorLi
 			"local.storageClass cannot change after creation")
 		allErrs = append(allErrs, err)
 	}
+	// annotation vertica.com/use-client-proxy cannot change after creation
+	if v.Annotations[vmeta.UseVProxyAnnotation] != oldObj.Annotations[vmeta.UseVProxyAnnotation] {
+		prefix := field.NewPath("metadata").Child("annotations")
+		errMsg := fmt.Sprintf("annotation %s cannot change after creation", vmeta.UseVProxyAnnotation)
+		err := field.Invalid(prefix.Key(vmeta.UseVProxyAnnotation),
+			v.Annotations[vmeta.VClusterOpsAnnotation],
+			errMsg)
+		allErrs = append(allErrs, err)
+	}
+	// proxy.image cannot change after creation
+	if v.Spec.Proxy.Image != oldObj.Spec.Proxy.Image {
+		err := field.Invalid(field.NewPath("spec").Child("proxy").Child("image"),
+			v.Spec.Proxy.Image,
+			"proxy.image cannot change after creation")
+		allErrs = append(allErrs, err)
+	}
```

Comment:
Move this to a separate function.

Reply:
Moved in https://github.com/vertica/vertica-kubernetes/pull/1000/commits/9812f4d46783cde7952b7b700df861aef5db6d88